cli/lib/kontena/cli/spinner.rb

```diff
@@ -168,6 +168,7 @@ def self.spin(msg, &block)
               Kernel.puts msg
             end
           end
+          Thread.main['spinners'].delete(spin_thread)
```

Comment:
If we `delete` the `spin_thread` from `Thread.main['spinners']` here in the ensure, then what does the `Thread.main['spinners'].pop` later on after the `ensure` block do... because if nothing gets raised, then this will run both the `Thread.main['spinners'].delete(spin_thread)` and `Thread.main['spinners'].pop`?

Reply:
Yeah, it does both... the `pop` is a no-op if the `Thread.main['spinners']` is empty: 

```
 [done] Triggering deployment of stack stop-signal     
delete #<Thread:0x000000021da7c0@/home/kontena/kontena/kontena/cli/lib/kontena/cli/spinner.rb:96 dead> from Thread.main['spinners']: [#<Thread:0x000000021da7c0@/home/kontena/kontena/kontena/cli/lib/kontena/cli/spinner.rb:96 dead>]
pop from Thread.main['spinners']: []
```

I'm guessing this would be relevant for nested spinners, though... where in the CLI are those actually used? I couldn't find any with a quick look around.

Reply:
Hopefully not anywhere. There was one in the master deploy wizard but I think it has been removed.

Reply:
Could this code be simplified to not be broken in complex cases that are not actually used? :/

Reply:
The whole spinner.rb should be buried. Very deep.

Reply:
![Nuke it from orbit, it's the only way to be sure](http://i.imgur.com/B51Emut.jpg)

Reply:
Dunno how the nested spinners are supposed to work, but testing by changing the `kontena stack install ` -> `Installing dependency ...` into a spinner, it actually behaves alright with this buggy change, at least without errors:

```
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo .. \
 [done] Creating stack deptest-foo      
 [done] Triggering deployment of stack deptest-foo     
 [done] Waiting for deployment to start     
 [done] Deploying service redis     
 [done] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo     
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar .. \
 [done] Creating stack deptest-bar      
 [done] Triggering deployment of stack deptest-bar     
 [done] Waiting for deployment to start     
 [done] Deploying service redis     
 [done] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar     
 [done] Creating stack deptest      
 [done] Triggering deployment of stack deptest     
 [done] Waiting for deployment to start     
 [done] Deploying service redis     
```

In fact, I maybe like the "broken" output with the delete+pop in this PR more than the "working" output before this PR?

```
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo .. \
 [done] Creating stack deptest-foo      
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo
 [done] Triggering deployment of stack deptest-foo     
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo
 [done] Waiting for deployment to start     
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo .. |
 [done] Deploying service redis     
 [done] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-foo     
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar .. \
 [done] Creating stack deptest-bar      
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar
 [done] Triggering deployment of stack deptest-bar     
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar
 [done] Waiting for deployment to start     
 [....] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar .. \
 [done] Deploying service redis     
 [done] Installing dependency /home/kontena/kontena/kontena/cli/test/deptest-child.yaml as deptest-bar     
 [done] Creating stack deptest      
 [done] Triggering deployment of stack deptest     
 [done] Waiting for deployment to start     
 [done] Deploying service redis     
```